cirq/ops/linear_combinations.py

```diff
@@ -315,6 +316,71 @@ def copy(self) -> 'PauliSum':
         factory = type(self)
         return factory(self._linear_dict.copy())
 
+    def expectation(self,
+                    state: np.ndarray,
+                    qubit_map: Optional[Dict[raw_types.Qid, int]]=None
+                    ) -> float:
+        """Evaluate the expectation value of this PauliSum.
+
+        Compute the expectation value of this Hermitian PauliSum with respect to
+        an array representing either a wavefunction or density matrix.
+
+        See `PauliString.expectation` for input requirements.
+
+        Args:
+            state: An array representing a wavefunction or density matrix.
+            qubit_map: A map from all qubits used in this PauliSum to the
+            indices of the qubits that `state` is defined over.
+
+        Returns:
+            The expectation value of the input state.
+
+        Raises:
+            NotImplementedError if any term in this PauliSum is non-Hermitian.
+            ValueError if the shape of `state` is neither `(2 ** n,)`
+                or `(2 ** n, 2 ** n)`.
+        """
+        if any([abs(p.coefficient.imag) > 0.0001 for p in self]):
+            raise NotImplementedError(
+                "Cannot compute expectation value of a non-Hermitian "
+                "PauliString <{}>. Coefficient must be real.".format(self))
+
+        if qubit_map is None:
+            qubit_map = {q: i for i, q in enumerate(self.qubits)}
```

Comment:
Inferring this value may cause user error. You run a simulation, take the output, pass it into the expectation... oops, the pauli sum had only a subset of the qubits.

At the very least this should be limited to the case where the number of qubits in the state is equal to the number of qubits in the operator.